app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -12,7 +12,7 @@ module HostManagedExtensions
         alias_method_chain :smart_proxy_ids, :katello
 
         has_one :content_host, :class_name => "Katello::System", :foreign_key => :host_id,
-                               :dependent => :restrict, :inverse_of => :foreman_host
+                               :dependent => :restrict_with_error, :inverse_of => :foreman_host
```

Comment:
It's a bit confusing but `ActiveRecord::DeleteRestrictionError` is an exception not an error. The option `restrict_with_error` sets an error on the object (e.g. Host.errors) while `restrict_with_exception` raises an exception (`ActiveRecord::DeleteRestrictionError`). Here's an example to illustrate this point:

Rails 3 `:dependent => :restrict`:
```
> u = User.create!(:name => "Test", :roles => [Role.first])
=> #<User id: 1, name: "Test", age: nil, created_at: "2015-12-10 19:25:51", updated_at: "2015-12-10 19:25:51">
> u.destroy
ActiveRecord::DeleteRestrictionError: Cannot delete record because of dependent roles
```

Rails 4 `:dependent => :restrict_with_error`
```
> u = User.create!(:name => "Test", :roles => [Role.first])
=> #<User id: 1, name: "Test", age: nil, created_at: "2015-12-10 19:25:51", updated_at: "2015-12-10 19:25:51">
> u.destroy
=> false
> u.errors
=> #<ActiveModel::Errors:0x007fbcd13c76d0 @base=#<User id: 2, first_name: "Bob1", last_name: nil, created_at: "2015-12-08 09:51:49", updated_at: "2015-12-08 09:51:49">, @messages={:base=>["Cannot delete record because dependent roles exist"]}>
```

Rails 4 `:dependent => :restrict_with_exception`
```
> u = User.create!(:name => "Test", :roles => [Role.first])
=> #<User id: 1, name: "Test", age: nil, created_at: "2015-12-10 19:25:51", updated_at: "2015-12-10 19:25:51">
> u.destroy
ActiveRecord::DeleteRestrictionError: Cannot delete record because of dependent roles
```